docs/modules/angular/pages/cookbooks/vanilla-to-onecx/migrate-to-onecx.adoc

```diff
@@ -0,0 +1,317 @@
+  },
+  shared: share({
+    '@angular/core': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/common': { requiredVersion: 'auto', includeSecondaries: { skip: ['@angular/common/http/testing'] } },
+    '@angular/common/http': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/forms': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/platform-browser': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/router': { requiredVersion: 'auto', includeSecondaries: true },
+    rxjs: { requiredVersion: 'auto', includeSecondaries: true },
+    '@onecx/angular-integration-interface': { requiredVersion: 'auto', includeSecondaries: true },
+    '@onecx/angular-utils': { requiredVersion: 'auto', includeSecondaries: true },
+    '@onecx/angular-webcomponents': { requiredVersion: 'auto', includeSecondaries: true },
+  }),
+})
+config.devServer = { allowedHosts: 'all' }
+const plugins = config.plugins.filter((plugin) => !(plugin instanceof ModifyEntryPlugin))
+module.exports = {
+  ...config,
+  plugins,
+  output: { uniqueName: 'YOUR_MFE_NAME', publicPath: 'auto' },
+  experiments: { ...config.experiments, topLevelAwait: true },
+  optimization: { runtimeChunk: false, splitChunks: false }
+}
+
+To allow for a custom Webpack configuration in production environments it's a good practice to additionally create a `webpack.prod.js` file that will extend the base config:
+[source, typescript]
+module.exports = require('./webpack.config')
+
+If necessary, this file can be used to overwrite or extend the contents of the base config with production specific values.
+
+=== Add environment file `env.json`
+During the loading of an app, OneCX will try to load it's environment file. To avoid any errors, a file called `env.json` should be accessible at http://localhost:4200/env.json. This can be achieved by adding the respective file to the assets or public folder of the application. The file can be used to pass environment variables to OneCX. For now it can just contain an empty object (`{}`).
+
+=== Verify that the app is exposing a remote entry point
+After completing all steps listed above, the application should be ready to be used as a microfrontend in the OneCX platform. After starting the application with its start command (e.g. `npm start`), the applications entrypoint file (remoteEntry.js) should be accessible at http://localhost:4200/remoteEntry.js.
+
+== Standalone Mode
+After completing all steps mentioned in this document, a previously existing "vanilla" Angular application should be runnable as a microfrontend in OneCX. If the app additionally has to support running in standalone (e.g. in local development), a few additional steps are necessary. All steps required for running an app in standalone are covered in the separate xref:cookbooks/enable-standalone/enable-standalone-mode.adoc[Enable OneCX app to run in standalone] cookbook.
+
+== Troubleshooting
```

Comment:
There could be a section referencing documentation of webcomponents and @onecx/angular-webcomponents so users know where to look in case they want to find out more about the migration tools.